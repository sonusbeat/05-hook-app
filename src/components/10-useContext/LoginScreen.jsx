import { useContext } from "react";
import UserContext from "./UserContext";

const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  return (
    <main>
      <h1 className="text-center">Login Screen</h1>
      <hr/>
      <button
        className="btn btn-primary"
        onClick={ () => setUser({ id: 123, name: "Daniel" }) }
      >
        Login
      </button>
    </main>
  );
}

export default LoginScreen;