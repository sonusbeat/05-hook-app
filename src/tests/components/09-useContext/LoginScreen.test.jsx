import { mount } from 'enzyme';
import UserContext from '../../../components/10-useContext/UserContext';
import LoginScreen from '../../../components/10-useContext/LoginScreen';

describe('Pruebas en <LoginScreen />', () => {

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('Deberia coincidir con el Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de ejecutar el setUser con el argumento esperado', () => {
    wrapper.find("button").prop("onClick")();

    expect(setUser).toHaveBeenCalledWith({ id: 123, name: "Daniel" });
  });

});
