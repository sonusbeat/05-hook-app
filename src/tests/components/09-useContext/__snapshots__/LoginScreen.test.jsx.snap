// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Deberia coincidir con el Snapshot 1`] = `
<main>
  <h1
    className="text-center"
  >
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</main>
`;
